dmscripts/upload_dos_opportunities_email_list.py

```diff
@@ -0,0 +1,46 @@
+from dmscripts.helpers.supplier_data_helpers import SupplierFrameworkData
+
+
+def find_user_emails(supplier_users, services):
+    """Return email addresses for any supplier who has a service in services."""
+    email_addresses = []
+    for service in services:
+        for user in supplier_users.get(service['supplierId'], []):
+            email_addresses.append(user['email address'])
+    return email_addresses
```

Comment:
Fair enough in terms of it may be easier to code a mistake but I'd say all that matters is you have correct test coverage and your tests pass